user_io.cpp

```diff
@@ -2504,7 +2508,10 @@ int user_io_file_tx(const char* name, unsigned char index, char opensave, char m
 	user_io_set_index(index);
 
 	int len = strlen(f.name);
-	char *p = f.name + len - 4;
+	char *p = strrchr(f.name, '.');
+	if (p == 0) {
+	    p = f.name + len - 4;
```

Comment:
Should this be `p = f.name + len` ?

Reply:
That could work. I thought reverting to the old behaviour would be enough.

There's also an edge case where a directory has `.` in it, but the file itself has no extension. I'm not handling that case at all.

If you feel that should be done and send NUL x 4 to the core (which is what mine does at the moment), I'm okay with that change.

Reply:
I did this, I guess it makes sense. 👍